### ISSUE TITLE
Drag & Drop folder to app icon should open the bash in folder

### ISSUE THREAD

**Issue by crispossa** [vercel/hyper#287]
In Terminal.app, when I drag & drop some folder to the app icon, it make cd [folder dropped] for me and it's very useful.

Such feature not work in HyperTerm.

Any plans to add this? #

**Comment by leo** [comment]
My god, this feature is so cool. We definitely need it in HyperTerm! Might be a great idea though to built it as a plugin and add it to the default configuration.

**Comment by timneutkens** [comment]
I agree with @leo would be a cool feature, will have a go at this.. shouldn't be too hard right :wink:

**Comment by startupfoundry** [comment]
I agree with @crispossa. This is such an ingrained part of my workflow that I had to switch back to iTerm 2. I really hope this gets added so I can give Hyperterm another spin!

**Comment by crispossa** [comment]
Unfortunately, I don't think this can be achieved with just a plugin, since this feature touches in the OS X API (Open With...). But, I can be wrong....

Cheers.

**Comment by timneutkens** [comment]
It is possible with electron (try vscode or atom, they both support it) just got to figure out how to enable dropping on the icon in electron. From what I can see here it might have to be in the core of hyperterm because of this: https://discuss.atom.io/t/how-to-handle-file-when-dragged-on-to-application-icon/12799/5

**Comment by timneutkens** [comment]
Also as a side note, it's also not possible to drop a file into the current tab. iTerm will automatically insert the file path at the cursor when you do that.

**Comment by jrdnbwmn** [comment]
+1

I use this all the time in Terminal.

**Comment by lordgiotto** [comment]
There are some pull request regarding this issue:
- #307 Enables file drop to dock icon
- #276 Handle the path passed by the dropped file and "cd" to the directory (or execute it if it's a file)

**Comment by CodeTheory** [comment]
@lordgiotto Is this issue still valid?

**Comment by lordgiotto** [comment]
@CodeTheory 
#307 was merged, so is possible to drop files on the dock icon.
#329 Is not merged yet, so for now dropping a file on a dock icon does nothing.

So this issue is not fixed yet :)

**Comment by Ravenna** [comment]
+1, loving Hyperterm just don't like typing paths to open a folder.

**Comment by lordgiotto** [comment]
Can be closed due #329 and #307 was merged :)

**Comment by crispossa** [comment]
Great job! Redownloading app!

**Comment by lordgiotto** [comment]
@crispossa You have to clone current master, the packaged version is not still updated to that feature: you have to wait the next release for this :)

**Comment by crispossa** [comment]
I know. Thx.

**Comment by marcotollini** [comment]
Hi @lordgiotto,

I've installed the latest canary version (2.0.2) but I can't drag and drop yet. 
I'm running Mac OS Sierra.
Do you have any suggestion?

Thanks!

**Comment by katiasmet** [comment]
@lordgiotto I have the same problem. I have version 1.4.8 and Mac OS Sierra but can't drag and drop.

**Comment by wolfgangrittner** [comment]
@lordgiotto same here, I'm running 1.4.8 on Sierra and dropping a folder (from finder, or any other source) on the dock icon does nothing.

**Comment by albinekb** [comment]
Looks like we have a regression here, @chabou could this have been accidentally removed in 2.x? 🤔

**Comment by chabou** [comment]
We have some feedbacks from 1.4.8 too.
We maybe have some old regression but not related to 2.X imo

**Comment by johnhearfield** [comment]
Could someone confirm progress on this? I recently updated to Hyper 3.0.0 (stable) and doesnt seem to be working for either Dock icon or drag and drop on the active window?

**Comment by riomarmccartney** [comment]
I'm having exactly the same problems since updating to v3, I can't drag and drop directories into the hyper interface anymore. 

<details><summary>This is my hyper config .hyper.js </summary>
<p>


```
// Future versions of Hyper may add additional config options,
// which will not automatically be merged into this file.
// See https://hyper.is#cfg for all currently supported options.

module.exports = {
  config: {
    // Choose either "stable" for receiving highly polished,
    // or "canary" for less polished but more frequent updates
    updateChannel: 'stable',

    // default font size in pixels for all tabs
    fontSize: 12,

    // font family with optional fallbacks
    fontFamily: 'Menlo, "DejaVu Sans Mono", Consolas, "Lucida Console", monospace',

    // terminal cursor background color and opacity (hex, rgb, hsl, hsv, hwb or cmyk)
    cursorColor: 'rgba(248,28,229,0.5)',

    // `BEAM` for |, `UNDERLINE` for _, `BLOCK` for █
    cursorShape: 'BLOCK',

    // set to true for blinking cursor
    cursorBlink: false,

    // color of the text
    foregroundColor: '#fff',

    // terminal background color
    backgroundColor: '#000',

    // border color (window, tabs)
    borderColor: '#333',

    // custom css to embed in the main window
    css: '',

    // custom css to embed in the terminal window
    termCSS: '',

    // set to `true` (without backticks) if you're using a Linux setup that doesn't show native menus
    // default: `false` on Linux, `true` on Windows (ignored on macOS)
    showHamburgerMenu: '',

    // set to `false` if you want to hide the minimize, maximize and close buttons
    // additionally, set to `'left'` if you want them on the left, like in Ubuntu
    // default: `true` on windows and Linux (ignored on macOS)
    showWindowControls: '',

    // custom padding (css format, i.e.: `top right bottom left`)
    padding: '12px 14px',

    // the full list. if you're going to provide the full color palette,
    // including the 6 x 6 color cubes and the grayscale map, just provide
    // an array here instead of a color map object
    colors: {
      black: '#000000',
      red: '#ff0000',
      green: '#33ff00',
      yellow: '#ffff00',
      blue: '#0066ff',
      magenta: '#cc00ff',
      cyan: '#00ffff',
      white: '#d0d0d0',
      lightBlack: '#808080',
      lightRed: '#ff0000',
      lightGreen: '#33ff00',
      lightYellow: '#ffff00',
      lightBlue: '#0066ff',
      lightMagenta: '#cc00ff',
      lightCyan: '#00ffff',
      lightWhite: '#ffffff'
    },

    verminal: {
      fontFamily: '"Space Mono"',
      fontSize: 12
    },
    // the shell to run when spawning a new session (i.e. /usr/local/bin/fish)
    // if left empty, your system's login shell will be used by default
    //
    // Windows
    // - Make sure to use a full path if the binary name doesn't work
    // - Remove `--login` in shellArgs
    //
    // Bash on Windows
    // - Example: `C:\\Windows\\System32\\bash.exe`
    //
    // Powershell on Windows
    // - Example: `C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe`
    shell: '',

    // for setting shell arguments (i.e. for using interactive shellArgs: ['-i'])
    // by default ['--login'] will be used
    shellArgs: ['--login'],

    // for environment variables
    env: {},

    // set to false for no bell
    bell: 'SOUND',

    // if true, selected text will automatically be copied to the clipboard
    copyOnSelect: false

    // if true, on right click selected text will be copied or pasted if no
    // selection is present (true by default on Windows)
    // quickEdit: true

    // URL to custom bell
    // bellSoundURL: 'http://example.com/bell.mp3',

    // for advanced config flags please refer to https://hyper.is/#cfg
  },

  // a list of plugins to fetch and install from npm
  // format: [@org/]project[#version]
  // examples:
  //   `hyperpower`
  //   `@company/project`
  //   `project#1.0.1`
  plugins: [
    "verminal",
    "hyper-statusline",
    "hyper-search"
  ],

  // in development, you can create a directory under
  // `~/.hyper_plugins/local/` and include it here
  // to load it and avoid it being `npm install`ed
  localPlugins: [],

  keymaps: {
    // Example
    // 'window:devtools': 'cmd+alt+o',
  }
};

```

</p>
</details>

**Comment by Scott-Fischer** [comment]
> I'm having exactly the same problems since updating to v3, I can't drag and drop directories into the hyper interface anymore.
> 
> This is my hyper config .hyper.js
> Edit: Still the same after a clean re-install -> My config shouldn't have caused this issue.

Same here! It used to work perfectly until I updated to v3. Now I'm stuck typing full paths since no drag+drop functionality works.

Any ideas why this was broken with the V3 release?

**Comment by HTMLangdon** [comment]
Hack: 

If you have Spotlight or Alfred 3 (I use the ladder)  
-click + initialize the drag folder
- holding your left mouse click down, open spotlight/alfred (CMD+Space)
- drop file into the search text - it will add the full path.
- copy/paste boom bing.

you can also drop it in browser url text box, and probably plenty of others.... anyway.. stupid but effective lo

**Comment by henryz00** [comment]
Any updates on this issue?

**Comment by bet4it** [comment]
@henryz00 
Use [v3.1.0-canary.4](https://github.com/zeit/hyper/releases/tag/v3.1.0-canary.4)

**Comment by LabhanshAgrawal** [comment]
Fixed in #4661
